src/services/codefixes/helpers.ts

```diff
@@ -56,15 +56,21 @@ namespace ts.codefix {
         const type = checker.getWidenedType(checker.getTypeOfSymbolAtLocation(symbol, enclosingDeclaration));
         const optional = !!(symbol.flags & SymbolFlags.Optional);
         const ambient = !!(enclosingDeclaration.flags & NodeFlags.Ambient);
+        const isSingleQuote = preferences.quotePreference === "single";
 
         switch (declaration.kind) {
             case SyntaxKind.PropertySignature:
             case SyntaxKind.PropertyDeclaration:
-                const typeNode = checker.typeToTypeNode(type, enclosingDeclaration, /*flags*/ undefined, getNoopSymbolTrackerWithResolver(context));
+                const propName = isStringLiteral(name)
+                    ? createLiteral(name.text, /*isSingleQuote*/ isSingleQuote)
+                    : name;
+                const typeNode = type.isStringLiteral()
+                    ? createLiteralTypeNode(createLiteral(type.value, /*isSingleQuote*/ isSingleQuote))
+                    : checker.typeToTypeNode(type, enclosingDeclaration, /*flags*/ undefined, getNoopSymbolTrackerWithResolver(context));

```

Comment:
This is not enough, since in interface if there is property `fooBar: { x: 'stringliteral' }` it wont get correctly quoted. We need to pass this to `typeToTypeNode` rather than doing it here.

Reply:
@sheetalkamat thanks for the feedback. will check.